account_payment_mode_default_account/tests/test_account_payment_mode_default_account.py

```diff
@@ -3,26 +3,118 @@
+                ),
+            ],
+            limit=1,
+        ),
+        # Override default_account_receivable and default_account_payable to use the stored accounts
+        "default_account_receivable": cls.env["res.partner"]
+        ._fields["property_stored_account_receivable_id"]
+        .get_company_dependent_fallback(cls.env["res.partner"].with_company(company)),
+        # End of the override
+        "default_account_payable": AccountAccount.search(
+            [*account_company_domain, ("account_type", "=", "liability_payable")],
+            limit=1,
+        ),
+        "default_tax_account_receivable": company.account_purchase_tax_id.tax_group_id.tax_receivable_account_id,
+        "default_tax_account_payable": company.account_sale_tax_id.tax_group_id.tax_payable_account_id,
+        "default_account_assets": AccountAccount.search(
+            [*account_company_domain, ("account_type", "=", "asset_fixed")], limit=1
+        ),
+        "default_account_deferred_expense": AccountAccount.search(
+            [*account_company_domain, ("account_type", "=", "asset_current")], limit=1
+        ),
+        "default_account_deferred_revenue": AccountAccount.search(
+            [*account_company_domain, ("account_type", "=", "liability_current")],
+            limit=1,
+        ),
+        "default_account_tax_sale": company.account_sale_tax_id.mapped(
+            "invoice_repartition_line_ids.account_id"
+        ),
+        "default_account_tax_purchase": company.account_purchase_tax_id.mapped(
+            "invoice_repartition_line_ids.account_id"
+        ),
+        "default_journal_misc": cls.env["account.journal"].search(
+            [*journal_company_domain, ("type", "=", "general")], limit=1
+        ),
+        "default_journal_sale": cls.env["account.journal"].search(
+            [*journal_company_domain, ("type", "=", "sale")], limit=1
+        ),
+        "default_journal_purchase": cls.env["account.journal"].search(
+            [*journal_company_domain, ("type", "=", "purchase")], limit=1
+        ),
+        "default_journal_bank": cls.env["account.journal"].search(
+            [*journal_company_domain, ("type", "=", "bank")], limit=1
+        ),
+        "default_journal_cash": cls.env["account.journal"].search(
+            [*journal_company_domain, ("type", "=", "cash")], limit=1
+        ),
+        "default_journal_credit": cls.env["account.journal"].create(
+            {
+                "name": "Credit Journal",
+                "type": "credit",
+                "code": "CCD1",
+                "company_id": company.id,
+            }
+        ),
+        "default_tax_sale": company.account_sale_tax_id,
+        "default_tax_purchase": company.account_purchase_tax_id,
+    }
+
+
+account_common.AccountTestInvoicingCommon.collect_company_accounting_data = classmethod(
+    _patched_collect_company_accounting_data
+)
 
 
 class TestAccountPaymentModeDefaultAccount(TransactionCase):
```

Comment:
you can try using BaseCommon class to inherit for the test instead of TransactionCase